spyder/plugins/editor/widgets/editor.py

```diff
@@ -1496,11 +1496,19 @@ def has_filename(self, filename):
             The self.data index for the filename.  Returns None
             if the filename is not found in self.data.
         """
-        fixpath = lambda path: osp.normcase(osp.realpath(path))
-        for index, finfo in enumerate(self.data):
-            if fixpath(filename) == fixpath(finfo.filename):
-                return index
-        return None
+        data_filenames = self.get_filenames()
+        try:
+            # Try finding without calling the slow realpath
+            return data_filenames.index(filename)
+        except ValueError:
+            def fixpath(path):

```

Comment:
I think there should be a fix where all the filenames are validated before being used in the editor. But this is a bit out of scope for this PR.

Reply:
I understand that, but I meant that I will prefer to not have nested functions because usually they are hard to follow and in this particular case you don't absolutely need the nested function there.

Reply:
I'll fix that